### PR TITLE
add `goal protocols`

### PR DESCRIPTION
**Pull request by algobolson**
add a goal command to dump all known protocols as json. Useful for other tools to import (Indexer).

### CODE REVIEW

File: cmd/goal/commands.go [hunk 1]
```diff
@@ -220,6 +221,16 @@ var reportCmd = &cobra.Command{
 	},
 }
 
+var protoCmd = &cobra.Command{
+	Use:   "protocols",
+	Short: "",
+	Long:  "Dump known consensus protocols as json to stdout.",
```
Comment (tsachiherman) [inline cmd/goal/commands.go:227]: could you rephrase it to "Dump built-in consensus.." ?
We do support dynamically loaded consensus protocols, so saying "built-in" would better describe these.

Comment (algobolson) [reply cmd/goal/commands.go:227]: how about "standard",  "Dump standard consensus protocols ..."

Comment (tsachiherman) [reply cmd/goal/commands.go:227]: Sounds good

File: cmd/goal/commands.go [hunk 2]
```diff
@@ -220,6 +221,16 @@ var reportCmd = &cobra.Command{
...
+	Args:  validateNoPosArgsFn,
+	Run: func(cmd *cobra.Command, args []string) {
+		os.Stdout.Write(protocol.EncodeJSON(config.Consensus))
```
Comment (pzbitskiy) [inline cmd/goal/commands.go:230]: mb also `-o` to dump as msgp

Comment (tsachiherman) [reply cmd/goal/commands.go:230]: I think that @pzbitskiy is a good idea, but let's finish this one first and consider to do the other in a separate PR.
